README.md

```diff
@@ -1,61 +1,46 @@
 # SecretHub Cross Language Golang Client<sup>**experimental**</sup>
 
-`secrethub-xgo` wraps the `secrethub-go` client with `cgo` exported functions so it can be called from other languages, e.g. C, Python, Ruby, NodeJS, and Java. 
+`secrethub-xgo` wraps the `secrethub-go` client with `cgo` exported functions so it can be called from other languages, e.g. C, C#, Python, Ruby, NodeJS, and Java. 
 
-For a proof of concept of how a client writting in another language can use `secrethub-xgo`, see the `clients/` folder. Right now it only contains a Java example, but more will be added in the future.
+At the moment, we provide the library for C# usage. 
 
-Make sure you build the `secrethub-xgo` client first before using the example client code. 
+In order to make use of the package, you will need to have the following installed:
+ - [.NET Core](https://docs.microsoft.com/en-gb/dotnet/core/install/)
+ - [Golang](https://golang.org/doc/install)
 
-To build the `secrethub-xgo` c-shared object, run the `build.sh` script. It will place the build in a `linux` or `darwin` folder, depending on your OS. 
+## Create and use the package
 
-You'll need to have [Golang](https://golang.org/doc/install) installed. 
+First, you have to make the NuGet Package and use it in your project. To do so, Here are the steps:
+1. execute `make nupkg` from the Makefile
+2. Go to your .NET project and run the following command: `dotnet add package SecretHub -s <path_to_your_secrethub-xgo_repo>`.
+3. Since our library uses Json.Net, you also need to add this dependency to your project. To do so, run the following command: `dotnet add package Newtonsoft.Json`.
```

Comment:
Can't we avoid this by using a `.nuspec` file? 

https://docs.microsoft.com/en-us/nuget/reference/nuspec#dependencies-element
https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#packing-using-a-nuspec

Reply:
Packing using the `.nuspec` file is the same as packing with the `.csproj` in which we put the details for packing. Therefore the complexity remains the same, however we add an extra file, which is the `.nuspec`. After that, the user will still need to install the package to the desired project. Therefore, I am leaving as it is.
To explain better, the `make nupkg` command generates the required files using swig, in the end resulting in the library file: `libSecretHubXGO.so` for Linux and `SecretHubXGO.dll` for Windows. Then we run the command 'dotnet pack dotnet/secrethub.csproj' which will make the `.nupkg` file. This is the file the user need to access the library in C#. Therefore they need to install it: either from local build or from NuGet Gallery.

Reply:
So does that mean the dependency does not have to be installed manually if the SecretHub SDK is installed through the NuGet Gallery?

Reply:
I tried now and we don't need to manually install the dependency. At the beginning of trying the example, It failed until I manually added the dependency. Don't know why though. 
Now that it works, I will remove this step from the guide.